### PR TITLE
Rename the `Import` class to `ImportSource`

### PR DESCRIPTION
**Pull request by agarny**
Addresses issue #189.

### CODE REVIEW

**Comment by hsorby** [comment]
The OS X build slave appears to be having issues with it's connection to the internet.  This is under investigation and clearly not fixed, I thought this had been resolved but obviously I will need to dig further into why it is having problems.

You can try resending the request again, @nickerso knows how :).

File: src/api/libcellml/component.h [hunk 1]
```diff
@@ -41,10 +41,10 @@ class LIBCELLML_EXPORT Component: public ComponentEntity
      * Make this component an imported component by defining an import model
      * from which to extract the named component from.
      *
-     * @param imp The import from which the named component originates.
+     * @param import The import source from which the named component originates.
      * @param name The name of the component in the imported model to use.
      */
-    void setSourceComponent(const ImportPtr &imp, const std::string &name);
+    void setSourceComponent(const ImportSourcePtr &import, const std::string &name);
```
Comment (hsorby) [inline src/api/libcellml/component.h]: The parameter here should be named importSource.

**Comment by agarny** [comment]
Thanks for the explanation @hsorby. I have no idea how to resend the request again, so will wait for @nickerso.

File: src/api/libcellml/importsource.h [hunk 2]
```diff
@@ -22,23 +22,23 @@ limitations under the License.
 namespace libcellml {
 
 /**
- * @brief The Import class.
+ * @brief The ImportSource class.
  *
- * The import class is for encapsulating an external CellML Model.
+ * The import source class is for encapsulating an external CellML Model.
  */
```
Comment (hsorby) [inline src/api/libcellml/importsource.h]: We need to add more documentation here, @nickerso and I talked about this today I will try and add our thinking on this tomorrow.  I have forgotten and don't want to write the wrong thinking down.

File: src/api/libcellml/importsource.h [hunk 3]
```diff
@@ -22,23 +22,23 @@ limitations under the License.
...
+    ~ImportSource(); /**< Destructor */
+    ImportSource(const ImportSource &rhs); /**< Copy constructor */
+    ImportSource(ImportSource &&rhs); /**< Move constructor */
+    ImportSource& operator=(ImportSource m); /**< Assignment operator */
 
     /**
      * @brief Set the source Model.
      *
-     * Set the source Model that this import refers to.
+     * Set the source Model that this import source refers to.
```
Comment (hsorby) [inline src/api/libcellml/importsource.h]: Again we should add more documentation to make this clearer about what the source model is and what we are doing.

**Review by hsorby — COMMENTED** [review]
Hopefully @nickerso can clarify the points I have been vague on.

**Comment by agarny** [comment]
Hmm... it looks like the macOS slave is now behaving...

**Comment by hsorby** [comment]
Yes it is temperamental it usually works but sometimes the network access seems to fail for reasons unknown.

File: src/api/libcellml/importsource.h (hunk 2 above)
Comment (nickerso) [reply src/api/libcellml/importsource.h]: I'm not one to let being wrong get in the way of writing a long comment on github :) hopefully the following is not so wrong that it points people in the wrong direction...@hsorby and I might revise tomorrow.

The basic idea is that the `ImportSource` does two things: the first is it points to a URL (the source) from which a CellML model is expected to be obtainable; the second is that it acts as a container/set/handle for all imported entities which should be obtained from the same *instance* of that source model. This becomes important, for example, when you have sibling components imported from the same instance (variable connections maintained) or different instances (connections severed). As with @hsorby's comment and from #188, I think a bit more of an explanation about what an `ImportSource` is would be useful here. I also now think "encapsulating an external CellML model" is actually incorrect/misleading.

Maybe @hsorby and I can come up with a better docstring tomorrow based on the above and comments on #188...

**Comment by MichaelClerx** [comment]
Can we merge this or are we waiting for better docstrings?

**Comment by nickerso** [comment]
Better docstrings would be good. I'll try to get time with @hsorby today to propose something.

**Comment by agarny** [comment]
@nickerso's changes look good to me. So, @nickerso, if you are OK with my own changes, then feel free to merge this in (or someone else can merge this in if he's happy with all the changes).

**Review by nickerso — APPROVED** [review]

**Pull request merged**